### ISSUE TITLE
抽取textField的props为js, 以mixins形式使用

### ISSUE THREAD

**Issue by shijiansu** [museui/muse-ui#374]
在看textField的源代码, 发现已经写得很好.

有个小建议, 可以考虑将textField的props抽出出来作为一个单独的js, 以mixins形式使用.

暂时想到的理由有三,
1. 减少内部源代码的重复代码, 例如在selectField.vue中, 是有使用textField.vue的, 由于需要传递对应的参数基本一样, 所以在selectField.vue的props又重复selectField.vue的props.
2. 可以使其他组件同样支持textField的props的时候, 更加统一和一致.
3. 和第一点原因的类似, 由于其他开发人员也可能利用textField的slot去制作自己的组件(或者整合其他第三方的组件, 或扩展), 如果能够以mixins形式使用textField的props, 而且Muse UI有新版本更新的时候这些自定义的组件向前兼容性更好, 不用手动去修改自定义的组件的props.

谢谢.

**Comment by myronliu347** [comment]
确实是个好建议，这个一直是个痛点

**Comment by shijiansu** [comment]
谢谢考虑!

**Comment by myronliu347** [comment]
不过我觉得，这个也算了是vue模板的一边弊端，这么叠加的使用，每次都要写很多props

**Comment by shijiansu** [comment]
确实, 因为实际使用的时候很多Vue的组件其实不能完全开箱即用的, 所有还要进行扩展的时候就比较麻烦, 有时改的时候希望能够做到侵入, 但是有时也是尽量从源代码着手重用.  

另外Vue官方也没有对于组件重用性的一个标准的指引(best practice), 现在社区上也只能大家相互摸索了. mixins这个做法也算是后来Vue弄出来的一个面向对象的父类多继承的一种方式, 减少重复工作. 如果能够定义好结构比较好的对象继承树结构的话, 那还是可以的.

另外, 如果作者你还有精力的话, 可以考虑一下Muse UI的二次开发规范和指引. 这个可能也有点意思, 因为可以看到一些Issues是希望Muse UI提供更多的组件, 或者整合第三方组件的. 如果有二次开发规范和指引, 那么可以打造基于Muse UI的组件生态圈, 把一些功能丰富的组件(也有可能是客制化比较重的组件)给社区的人自己去写, 然后Muse UI这边的文档给出链接, 这样可以相对减少你的工作量和维护量, 同时也解耦(不会被新的社区组件影响原Muse UI, 也可以按需加载), 也可以使Muse UI组件更加适用于生产环境.

这个只是这几天学习使用Muse UI的一点看法, 如有不妥, 请多包容见谅, 谢谢!
